msaad.go

```diff
@@ -751,32 +756,58 @@ func numParallelFetchThreads(nItems int) int {
 // separating this step from the fetching of the AAD users
 func (m *MSAAD) populateAADRoles(users []*msaadUser) error {
 	nThreads := numParallelFetchThreads(len(users))
+	if m.parent.MSAAD.Config.Verbose {
+		m.parent.Log.Infof("MSAAD populateAADRoles started...%d\n", len(users))
+		m.parent.Log.Infof("MSAAD populateAADRoles : threads = %d\n", nThreads)
+	}
+
 	// partition 'users' into nThreads groups
 	threadGroups := make([][]*msaadUser, nThreads)
 	for i, u := range users {
 		t := i % nThreads
 		threadGroups[t] = append(threadGroups[t], u)
 	}
 	wg := sync.WaitGroup{}
-	errChan := make(chan error, nThreads)
-
+	var errGlobal error
 	startTime := time.Now()
+	i := 0
 	for _, threadGroupOuter := range threadGroups {
```

Comment:
Pretty sure you can just piggy back off this.
```suggestion
	for i, threadGroupOuter := range threadGroups {
```

Reply:
of course